load_tests/load_test.py

```diff
@@ -436,22 +441,36 @@ def publish_fluent_config_s3(input_logger):
     )
     return f'arn:aws:s3:::{bucket_name}/{OUTPUT_PLUGIN}-test/{PLATFORM}/fluent-{input_logger["name"]}.conf'
 
+
 # The following method is used to clear data after all tests run
-def delete_testing_data(session):
-    # Delete associated cloudwatch log streams
-    client = session.client('logs')
-    response = client.describe_log_streams(
-        logGroupName=os.environ['CW_LOG_GROUP_NAME']
-    )
-    for stream in response["logStreams"]:
-        client.delete_log_stream(
-            logGroupName=os.environ['CW_LOG_GROUP_NAME'],
-            logStreamName=stream["logStreamName"]
-        )
-    # Empty s3 bucket
-    s3 = session.resource('s3')
-    bucket = s3.Bucket(os.environ['S3_BUCKET_NAME'])
-    bucket.objects.all().delete()
+# Output should be "all", "cloudwatch", "firehose", "kinesis", or "s3"
+def delete_testing_data(session, output="all"):
+
+    if (output == "cloudwatch" or output == "all"):
+        # Delete associated cloudwatch log streams
+        if os.getenv('CW_LOG_GROUP_NAME') != None:
+            client = session.client('logs')
+            response = client.describe_log_streams(
+                logGroupName=os.getenv('CW_LOG_GROUP_NAME')
+            )
+            for stream in response["logStreams"]:
+                client.delete_log_stream(
+                    logGroupName=os.environ['CW_LOG_GROUP_NAME'],
+                    logStreamName=stream["logStreamName"]
+                )
+
+    if output == "all":
+        if os.getenv('S3_BUCKET_NAME') != None:
+            s3 = session.resource('s3')
+            bucket = s3.Bucket(os.environ['S3_BUCKET_NAME'])
+            bucket.objects.all().delete()
+
+    else:
+        if os.getenv('S3_BUCKET_NAME') != None:
+            s3 = session.resource('s3')
+            bucket = s3.Bucket(os.environ['S3_BUCKET_NAME'])
+            bucket.objects.filter(Prefix=(output.lower() + "-test/")).delete()
+
 
 def generate_daemonset_config(throughput):
```

Comment:
consider break this method into smaller one for readability  line 532 - 609 , it too long to read :)